### PR TITLE
Remove usage of Java assert in tests (#30220)

### PR DESCRIPTION
**Pull request by Captain1653**
References #30220

I removed assert almost everywhere, but some tests failed:
`ActorCreationTest.testWrongAnonymousInPlaceCreator` from [here](https://github.com/akka/akka/pull/30165#issuecomment-831126790)
`RemoteDeploymentDocTest.demonstrateDeployment` I have found

I didn't change them in this PR. I'm going to investigate the tests in my next PR. It can take some time and show some different problems (maybe).

Kind regards

### CODE REVIEW

File: akka-docs/src/test/java/jdocs/io/japi/IODocTest.java [hunk 1]
```diff
@@ -186,12 +189,12 @@ public void testConnection() {
 
         final Connected c1 = expectMsgClass(Connected.class);
         final Connected c2 = expectMsgClass(Connected.class);
-        assert c1.localAddress().equals(c2.remoteAddress());
-        assert c2.localAddress().equals(c1.remoteAddress());
+        assertTrue(c1.localAddress().equals(c2.remoteAddress()));
```
Comment (Captain1653) [inline akka-docs/src/test/java/jdocs/io/japi/IODocTest.java:192]: use assertTrue, because assertEquals(null,null) will be green.

**Comment by raboof** [comment]
OK TO TEST

File: akka-docs/src/test/java/jdocs/stream/StreamTestKitDocTest.java [hunk 2]
```diff
@@ -66,7 +67,7 @@ public void groupedPartOfInfiniteStream() throws Exception {
     final CompletionStage<List<Integer>> future =
         sourceUnderTest.take(10).runWith(Sink.seq(), system);
     final List<Integer> result = future.toCompletableFuture().get(3, TimeUnit.SECONDS);
-    assertEquals(result, Collections.nCopies(10, 2));
+    assertEquals(Collections.nCopies(10, 2), result);
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/stream/StreamTestKitDocTest.java:70]: :+1:

**Review by raboof — APPROVED** [review]
Thanks!

**Review by octonato — APPROVED** [review]
LGTM

Amazing that all those assertions were there, not being executed, but still none of the tests got broken when you refactored to effectively run the assertion. 👍

**Comment by octonato** [comment]
Oh, now I see your comment in the description. I knew about `ActorCreationTest`, but not about `RemoteDeploymentDocTest`. Still impressive that only two are broken.

**Pull request merged**